### ISSUE TITLE
EXC_BAD_ACCESS in MemoryBlock::write_bytes

### ISSUE THREAD

**Issue by kvark** [zakarumych/gpu-alloc#27]
See https://github.com/gfx-rs/wgpu/issues/1038
We are still figuring out a way to reproduce this.

**Comment by kvark** [comment]
Caught that in XCode debugger. Here is the block info:
```
self	gpu_alloc::block::MemoryBlock<alloc::sync::Arc<gfx_backend_metal::native::Memory> > *	0x7ffee0216068	0x00007ffee0216068
memory	alloc::sync::Arc<gfx_backend_metal::native::Memory>	
memory_type	unsigned int	1
props	gpu_alloc_types::types::MemoryPropertyFlags	
bits	unsigned char	'\x06'
offset	unsigned long	40108032
size	unsigned long	256
map_mask	unsigned long	0
mapped	core::sync::atomic::AtomicU8	
v	core::cell::UnsafeCell<unsigned char>	
value	unsigned char	'\x01'
flavor	gpu_alloc::block::MemoryBlockFlavor	
relevant	gpu_alloc::block::Relevant	
```
Unfortunately, `flavor` can't be expanded by XCode, though this would be very much needed here.
The interesting bit here is the backing memory:
```
data	gfx_backend_metal::native::Memory	
heap	gfx_backend_metal::native::MemoryHeap	
size	unsigned long	33554432
```

So somehow the native memory size is 33Mb, while the block offset is 40Mb. This doesn't make much sense.

**Comment by zakarumych** [comment]
I've added few debug assertions. Particularly before constructing memory block to return from sub-allocator that double-checks offset + size is in bounds of memory object.

This is not `Dedicated` as offset is not zero.
And seems to be not `Buddy`, as memory size is not power of two.

**Comment by zakarumych** [comment]
After reading `linear.rs` one more time I found what caused the issue.
Pretty stupid mistake https://github.com/zakarumych/gpu-alloc/commit/7f306854cb0bba8aa314b123f0a75213886f9154

**Comment by kvark** [comment]
Awesome, thank you!
Fixed by b807b5fb41dc1a102d5ed5f8506265192c7a1c5e
